externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/session/HiveSessionImpl.scala

```diff
@@ -63,7 +67,16 @@ class HiveSessionImpl(
       case TGetInfoType.CLI_SERVER_NAME => TGetInfoValue.stringValue("Hive")
       case TGetInfoType.CLI_DBMS_NAME => TGetInfoValue.stringValue("Apache Hive")
       case TGetInfoType.CLI_DBMS_VER => TGetInfoValue.stringValue(HiveVersionInfo.getVersion)
-      case TGetInfoType.CLI_ODBC_KEYWORDS => TGetInfoValue.stringValue("Unimplemented")
+      case TGetInfoType.CLI_ODBC_KEYWORDS =>
+        if (SemanticVersion(HiveVersionInfo.getVersion).isVersionAtLeast("3.0.0")) {
+          val keywords = invokeAs[String](
```

Comment:
prefer to use `Dyn*` classes in engine modules, the chain syntax is much clearer to reflect the API evolution

Reply:
also, please leave a JIRA ticket in the comments, in practice, venders tend to backport lots of patches to lower versions of Hive, given that, prefer to judge by methods present or not rather than the Hive runtime version